doc/proxy-injection.md

```diff
@@ -1,94 +1,66 @@
-    name: http
-  selector:
-    app: example
----
-apiVersion: extensions/v1beta1
-kind: Deployment
-metadata:
-  name: example
-spec:
-  replicas: 1
-  template:
-    metadata:
-      labels:
-        app: example
-      annotations:
-        pod.beta.kubernetes.io/init-containers: >
-          [{
-            "name": "proxy-init",
-            "image": "docker.io/istio/init:latest",
-            "imagePullPolicy": "Always",
-            "securityContext": { "capabilities" : { "add" : ["NET_ADMIN"] } }
-          }]
-    spec:
-      containers:
-      - name: app
-        image: docker.io/istio/app:latest
-	imagePullPolicy: Always
-        args:
-          - --port
-          - "8080"
-        ports:
-        - containerPort: 8080
-      - name: proxy
-        image: docker.io/istio/runtime:latest
-        imagePullPolicy: Always
-        securityContext:
-          runAsUser: 1337
-        args:
-          - proxy
-          - -s
-          - manager:8080
----
-```
+    kubectl get deployment -o yaml | istioutil inject -f - | kubectl apply -f -
 
-## Automatic injection
+`istioutil inject` will update
+the [PodTemplateSpec](https://kubernetes.io/docs/api-reference/v1/definitions/#_v1_podtemplatespec) in
+kubernetes Job, DaemonSet, ReplicaSet, and Deployment YAML resource
+documents. Support for additional pod-based resource types can be
+added as necessary.
+
+Unsupported resources are left unmodified so, for example, it is safe
+to run `istioutil inject` over a single file that contains multiple
+Service, ConfigMap, and Deployment definitions for a complex
+application.
 
-Istio's goal is transparent proxy injection into end-user deployments with minimal effort from the end-user. Ideally, a kubernetes admission controller would rewrite specs to include the necessary init and proxy containers before they are committed, but his currently requires upstreaming changes to kubernetes which we would like to avoid for now. Instead, it would be better if a dynamic plug-in mechanism existed whereby admisson controllers could be maintained out-of-tree. There is no platform support for this yet, but a proposal has been created to add such a feature (see [Proposal: Extensible Admission Control](https://github.com/kubernetes/community/pull/132/)).
+The Istio project is continually evolving so the low-level proxy
+configuration may change unannounced. When it doubt re-run `istioutil
+inject` on your original deployments.
 
-Long term istio automatic proxy injection is being tracked by [Kubernetes Admission Controller for proxy injection](https://github.com/istio/manager/issues/57).
+```
+$ istioutil inject --help
+Inject istio runtime into existing kubernete resources
+
+Usage:
+   inject [flags]
+
+Flags:
+      --discoveryPort int     Manager discovery port (default 8080)
+  -f, --filename string       Unmodified input kubernetes resource filename
+      --initImage string      Istio init image (default "docker.io/istio/init_debug:latest")
```

Comment:
The biggest concern is the size. Debug envoy build binary is over 150MB in size.